tests/integration_tests/functional/test_max_vcpus.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests scenario for microvms with max vcpus(32)."""
+import os
+import select
+from framework.state_machine import TestState
+
+MAX_VCPUS = 32
+
+
+class WaitCPUs(TestState):
+    """Initial state when we wait for the login prompt."""
+
+    def handle_input(self, microvm, input_char) -> TestState:
+        """Handle input and return next state."""
+        if self.match(input_char):
+            return TestFinished()
+        return self
+
+
+class TestFinished(TestState):
+    """Test complete and successful."""
+
+    def handle_input(self, microvm, input_char) -> TestState:
+        """Return self since the test is about to end."""
+        return self
+
+
+def test_max_vcpus(test_microvm_with_ssh):
+    """Test if all configured vcpus are brought up at boot."""
+    microvm = test_microvm_with_ssh
+    microvm.jailer.daemonize = False
+    microvm.spawn()
+
+    # Configure a microVM with 32 vCPUs and a serial console.
+    microvm.basic_config(vcpu_count=MAX_VCPUS,
+                         boot_args='console=ttyS0 reboot=k panic=1 pci=off')
+
+    microvm.start()
+
+    # Setup screen log
```

Comment:
Any reason for using the serial to check that the CPUs are online? I think the code could be simplified if using the SSH connection instead and just check the output of the command `cat /sys/devices/system/cpu/online`.

Reply:
I was aiming to implement something very simple and fast, but this turns out to take 300 s if the test fails. Otherwise in the case of a successful test, the validation happens before the guest  finished boot and that is faster than logging in via ssh.

Reply:
Indeed there is less code using the ssh connection.